### PR TITLE
GH-15989: Craft DT Example

### LINKED ISSUE #15989: DecisionTree pydocs updates

### PR DESCRIPTION
**Pull request by shaunyogeshwaran**
fixes #15989 
Crafted new examples for the DT algorithm to showcase the different parameters.

### CODE REVIEW

**Review by hannah-tillman — APPROVED** [review]

**Review by hannah-tillman — DISMISSED** [review]

File: h2o-py/h2o/estimators/decision_tree.py [hunk 1]
```diff
@@ -107,6 +107,21 @@ def ignore_const_cols(self):
...
 
         Type: ``bool``, defaults to ``True``.
+
+        :examples:
+
+        >>> import h2o
+        >>> from h2o.estimators import H2ODecisionTreeEstimator
+        >>> h2o.init()
+        >>> prostate = h2o.import_file("http://s3.amazonaws.com/h2o-public-test-data/smalldata/prostate/prostate.csv")
+        >>> target_variable = 'CAPSULE'
```
Comment (valenad1) [inline h2o-py/h2o/estimators/decision_tree.py:117]: Add some constant values.

prostate["const_1"] = 6

File: h2o-py/h2o/estimators/decision_tree.py [hunk 2]
```diff
@@ -122,6 +137,21 @@ def categorical_encoding(self):
...
         Type: ``Literal["auto", "enum", "one_hot_internal", "one_hot_explicit", "binary", "eigen", "label_encoder",
         "sort_by_response", "enum_limited"]``, defaults to ``"auto"``.
+
+        :examples:
+
+        >>> import h2o
+        >>> from h2o.estimators import H2ODecisionTreeEstimator
+        >>> h2o.init()
+        >>> prostate = h2o.import_file("http://s3.amazonaws.com/h2o-public-test-data/smalldata/prostate/prostate.csv")
+        >>> target_variable = 'CAPSULE'
```
Comment (valenad1) [inline h2o-py/h2o/estimators/decision_tree.py:147]: Define also `RACE` as categorical and use different encoding than default.

File: h2o-py/h2o/estimators/decision_tree.py [hunk 3]
```diff
@@ -122,6 +137,21 @@ def categorical_encoding(self):
 
         Type: ``Literal["auto", "enum", "one_hot_internal", "one_hot_explicit", "binary", "eigen", "label_encoder",
         "sort_by_response", "enum_limited"]``, defaults to ``"auto"``.
```
Comment (valenad1) [inline h2o-py/h2o/estimators/decision_tree.py:139]: Out of scope but I would mention that `auto` means `enum`.

File: h2o-py/h2o/estimators/decision_tree.py [hunk 4]
```diff
@@ -178,6 +223,21 @@ def min_rows(self):
...
+        >>> import h2o
+        >>> from h2o.estimators import H2ODecisionTreeEstimator
+        >>> h2o.init()
+        >>> prostate = h2o.import_file("http://s3.amazonaws.com/h2o-public-test-data/smalldata/prostate/prostate.csv")
+        >>> target_variable = 'CAPSULE'
+        >>> prostate[target_variable] = prostate[target_variable].asfactor()
+        >>> train, test = prostate.split_frame(ratios=[0.7])
+        >>> sdt_h2o = H2ODecisionTreeEstimator(model_id="decision_tree.hex",
+        ...                                    max_depth=5,
+        ...                                    min_rows=10)
```
Comment (valenad1) [inline h2o-py/h2o/estimators/decision_tree.py:238]: Use different number than default.

**Review by valenad1 — CHANGES_REQUESTED** [review]
One example is not working and I would use different number than default.

Otherwise, LGTM, Thank you!

**Review by hannah-tillman — APPROVED** [review]

**Review by wendycwong — APPROVED** [review]

**Comment by shaunyogeshwaran** [comment]
Hello @valenad1, I've made the requested changes 👍

**Review by valenad1 — APPROVED** [review]
Works for me. Thank you @shaunyogeshwaran

**Pull request merged**